### ISSUE TITLE
grayed out at very top of conky manager v2.73

### ISSUE THREAD

**Issue by hdawg12** [zcot/conky-manager2#28]
Hello, 
I"m using Kali linux v 2021.4 and I followed everything in the directions, but when I open conky-manager v2.73, I have most of the buttons grayed out at the top. I can't start any widgets.  Do you know what the problem is?  i have a screenshot attached

![conky_managerV2](https://user-images.githubusercontent.com/45344753/147537048-8df12ec2-618d-4fb7-a389-a306f3577a00.png)

**Comment by zcot** [comment]
I don't know.

Most of the icons at the top are not relevant anyway unless you have selected one of the configs below(even though I never experience these disabled icons, they just don't do anything if nothing is selected below).

Are you running as a root account? I'm not sure if there is some issue with that.

**Comment by hdawg12** [comment]
i have tried selecting something below, but it still grayed out.  do you think it could be some type of bug related to version 2021.4? Do i have to have LUA installed? if so where do i go to see if I do? If I don't , where or what command should I type to install it?

**Comment by hdawg12** [comment]
btw, yes I am running as a root account

**Comment by zcot** [comment]
There's probably some issue with that. Did you test it from the user account, not root?

**Comment by hdawg12** [comment]
no, I just have a root account. I would have to create a user acct and test it, but the it worked before when I was using it on my root acct.  Not sure why its not now.

**Comment by zcot** [comment]
I don't know if that's a problem or not, I never run the root gui account, which is also typically what everyone is doing on Debian/Ubuntu.

If it worked before then I would not suspect it to be a problem now. I don't know what has changed, nothing interesting related with conky-manager2 I would suspect.

I did do a little searching related to kali and conky and it looks like kali uses a newer version of lua where conky is internally failing related to the convert script, but you can easily just manually convert those older configs(which maybe all the default ones supplied with conky-manager2 are): https://github.com/brndnmtthws/conky/wiki/Configurations#converting-old-configs-to-110 And yes, you would need lua installed on the system to be able to do that, but it shouldn't have any issue with conky-manager2 itself, but of course you do not get anywhere anyway with broken configs.

**Comment by hdawg12** [comment]
thanks for the update.  Quick question.  Is there a recent upgrade in lua? In my lua directory I have 5.1, 5.2, 5.3

**Comment by zcot** [comment]
conky-manager2 doesn't run on lua.

One thing a user might do when experiencing a problem with a gui program, is to start the program from terminal, so maybe you can get some feedback as to what is going on.

In this case, since the manager is strictly a function of dealing with conky configs, I would want to check conky itself, is everything there functioning correctly in the first place? Can you run a custom config? I would specifically check one that is using lua. Hell, does the default one even work?

Work up to the problem. Let me know if there's something I can do that is broken with the manager.

**Comment by hdawg12** [comment]
Yes I was doing that.  I'm able to get the default conky working on the conky manager.  I had to change some things on the config to make it happen, but it worked using conky manager v2.
Does each widget come with a conky.config? Which file would I need to work on for each widget so I can try and make it functional with conky-managerV2? What would the 
![conky_managerv2](https://user-images.githubusercontent.com/45344753/148019602-de316ead-b1be-46a9-bf96-a0cd40bdd07a.png)
file be called?

**Comment by zcot** [comment]
If the config doesn't work with conky-manager then it doesn't work with conky itself. The actual order is that if it does not work with conky in the first place, then it also can not work using the conky-manager2.

If conky is broken because it tries to use an old-version config and fails when it comes to internally using its contained lua script then the same failure would be present in the manager. The manager only runs the config manually for you, using conky, it's only a tool of convenience, it is not doing anything that conky itself doesn't do, as far as running a config goes.

For example, on terminal: `cd ~/.conky/Gotham ; conky -c ~/.conky/Gotham/Gotham`
-running this old-version config will show the following lines in terminal:
```
conky: Syntax error (/home/zcot/.conky/Gotham/Gotham:1: '=' expected near 'yes') while reading config file. 
conky: Assuming it's in old syntax and attempting conversion.
```
-the next lines are the normal output because typically conky will succeed in the conversion, but when it fails then you get error message(s) instead.

On a system where Lua 5.3 is installed that conversion is going to fail because the original conversion script used the lua call for 'loadstring' which was deprecated and only lua 'load' is going to work. You would either need to use the latest version of conky, where the conversion script is fixed, or just download the fixed script and convert the config yourself.

As for the conky-manager2, like I mentioned, you'll have to figure it out, there's nothing I can do. You can run it from terminal to see if you get any useful error messages, I already mentioned that. Maybe you find something on the kali support forums about how to use conky? If you can get a functional conky setup then at least you can pinpoint a problem directly with conky-manager2.

**Comment by hdawg12** [comment]
do you have the link to the latest version of conky?

**Comment by zcot** [comment]
I see it there: https://startpage.com/row/search?q=do+you+have+the+link+to+the+latest+version+of+conky%3F&l=english
